frame/proxy/src/lib.rs

```diff
@@ -305,17 +311,23 @@ pub mod pallet {
 		/// Weight is a function of the number of proxies the user has (P).
 		/// # </weight>
 		/// TODO: Might be over counting 1 read
-		#[pallet::weight(T::WeightInfo::anonymous(T::MaxProxies::get().into()))]
-		pub fn anonymous(
+		#[pallet::weight(T::WeightInfo::proxied_keyless(T::MaxProxies::get().into()))]
+		pub fn proxied_keyless(
 			origin: OriginFor<T>,
 			proxy_type: T::ProxyType,
 			delay: T::BlockNumber,
 			index: u16,
+			maybe_keyless: Option<T::AccountId>,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			let anonymous = Self::anonymous_account(&who, &proxy_type, index, None);
-			ensure!(!Proxies::<T>::contains_key(&anonymous), Error::<T>::Duplicate);
+			let keyless = if let Some(account) = maybe_keyless {
+				ensure!(Self::is_vain_enough(&account), Error::<T>::NotKeyless);
+				account
```

Comment:
what? you allow people to specify the account? that for sure is a security hole.

Reply:
But is it though? I don't think it's _sure_, that's my main question here as this is based on an empirical assumption. There's gonna be a line to not cross when evaluating the "keylessness" of an account that might even get blurry at some point, but we can always be super conservative about it to stay on the safe side. The initial approach here is leaving part of the judgment to the implementer that should specify a safe `MaxVanity`. Perhaps a crypto-math expert can jump in and say with certainty where the safe line is? is it much different from the safety of a pallet id(e.g. `modlpy/trsry`)?

Reply:
No it is different thing. You allow people to pick any addresses that fits a requirement without many more complicated checking and that will allow reuse address attack etc. Allow anyone to pick their own address is almost always a bad idea.

Reply:
What would a more complicated checking look like? perhaps make the process more expensive/permissioned? an auction? should governance be like a registar that sells/leases the right to have those accounts? Not to sound stubborn but I think it's a feature too cool to not make the effort to make it work somehow. wouldn't `1AcaLaNetwork1111111111111111111111111111111L6H` be a nice account to have for a project?